src/fn_call.rs

```diff
@@ -760,3 +783,21 @@ pub trait EvalContextExt<'a, 'mir, 'tcx: 'a + 'mir>: crate::MiriEvalContextExt<'
         self.eval_context_mut().write_scalar(Scalar::from_int(0, dest.layout.size), dest)
     }
 }
+
+fn gen_random<'a, 'mir, 'tcx>(this: &mut MiriEvalContext<'a, 'mir, 'tcx>,
+                                 len: usize) -> Result<Vec<u8>, EvalError<'tcx>>  {
+
+    match this.machine.rng.as_ref() {
+        Some(rng) => {
+            let mut data = vec![0; len];
+            rng.borrow_mut().fill_bytes(&mut data);
+            Ok(data)
+        }
+        None => {
+            err!(Unimplemented(
+                "miri does not support random number generators in deterministic mode!
+                Use '-Zmiri-seed=<seed>' to enable random number generation".to_owned(),
```

Comment:
This is not entirely accurate, RNGs still work, just seeding them with system entropy does not.  Can you reword the message appropriately?  Something like "Miri does not support gathering system entropy in deterministic mode" or so.

Also, please add a warning that this will *not* give rise to cryptographically secure entropy, and people shouldn't run crypto stuff in Miri.